### PR TITLE
BOAC-93, /api/cohort/create and /api/cohort/delete/<code> with test coverage

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-93

Custom cohort CRUD with tests.

### CODE REVIEW

File: boac/api/cohort_controller.py [hunk 1]
```diff
@@ -15,47 +17,89 @@ def teams_list():
...
+@app.route('/api/cohort/delete/<code>', methods=['DELETE'])
+@login_required
+def delete_cohort_filter(code):
+    cohort_filter_id = int(code)
+    uid = current_user.get_id()
+    cohort = get_cohort_filter_owned_by(cohort_filter_id, uid)
+    if cohort:
+        authorized_user.delete_cohort_filter(cohort_filter_id)
+        if app.cache:
+            app.cache.delete(get_cache_key(code))
```
Comment (johncrossman) [inline boac/api/cohort_controller.py]: Werkzeug `delete` doc: http://werkzeug.pocoo.org/docs/0.12/contrib/cache/

File: boac/api/cohort_controller.py [hunk 2]
```diff
@@ -15,47 +17,89 @@ def teams_list():
...
+    # Cohort must exist and be owned by user
+    for cohort_filter in authorized_user.cohort_filters_owned_by(uid):
+        if cohort_filter_id == cohort_filter.id:
+            members = []
+            # Extract team codes from filter_criteria
+            filter_criteria = json.loads(cohort_filter.filter_criteria)
+            team_codes = []
+            for team_code in filter_criteria['teams']:
+                team = get_team_details(team_code)
+                members = members + team['members']
```
Comment (paulkerschen) [inline boac/api/cohort_controller.py]: @johncrossman Looks like Python does provide `+=` for lists too.

**Comment by paulkerschen** [comment]
@johncrossman Left one minor style comment, but looks good to me!

File: boac/api/cohort_controller.py (hunk 2 above)
Comment (johncrossman) [reply boac/api/cohort_controller.py]: Nifty! Amendment is in.

**Pull request merged**